src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -125,6 +125,21 @@ public StackTrace(StackFrame frame)
             _numOfFrames = 1;
         }
 
+        /// <summary>
+        /// Constructs a "fake" stack trace
+        /// </summary>
+        public StackTrace(IEnumerable<StackFrame> frames)
+        {
+            ArgumentNullException.ThrowIfNull(frames);
+
+            List<StackFrame> frameList = new List<StackFrame>();
```

Comment:
>List<StackFrame> frameList = new List<StackFrame>(); [](http://example.com/codeflow?start=12&length=52)

if it is worth it, we may avoid this allocation by enumerating the frames twice. The first enumeration calculate the count and second just fill the _stackFrames array.

Reply:
IMO we'd be fine optimizing for code simplicity in this case.